v-next/hardhat-network-helpers/src/internal/network-helpers/network-helpers.ts

```diff
@@ -0,0 +1,452 @@
+import type { BlockTag, NumberLike, SnapshotRestorer } from "../../types.js";
+import type { EthereumProvider } from "@ignored/hardhat-vnext/types/providers";
+
+import {
+  assertHardhatInvariant,
+  HardhatError,
+} from "@ignored/hardhat-vnext-errors";
+
+import { clearSnapshots } from "../../load-fixture.js";
+import {
+  assertHexString,
+  assertLargerThan,
+  assertTxHash,
+  assertValidAddress,
+} from "../assertions.js";
+import { toBigInt, toNumber, toRpcQuantity } from "../conversion.js";
+import { toPaddedRpcQuantity } from "../padding.js";
+
+import { Time } from "./time.js";
+
+export class NetworkHelpers {
+  readonly #provider: EthereumProvider;
+
+  public time: Time;
+
+  constructor(provider: EthereumProvider) {
+    this.#provider = provider;
+
+    this.time = new Time(this, provider);
```

Comment:
Needed because the `time` class uses methods from the network helpers class. It also goes in the opposite direction: the network helpers class uses methods from the time class.